src/python/src/grpc/_adapter/_c/__init__.pyx

```diff
@@ -0,0 +1,992 @@
+    credentials = ClientCredentials()
+    credentials.c_credentials = grpc.grpc_refresh_token_credentials_create(
+        json_refresh_token)
+    credentials.references.append(json_refresh_token)
+    return credentials
+
+  @staticmethod
+  def fake_transport_security():
+    credentials = ClientCredentials()
+    credentials.c_credentials = \
+        grpc.grpc_fake_transport_security_credentials_create()
+    return credentials
+
+  @staticmethod
+  def iam(authorization_token, authority_selector):
+    if isinstance(authorization_token, bytes):
+      pass
+    elif isinstance(authorization_token, basestring):
+      authorization_token = authorization_token.encode()
+    else:
+      raise TypeError("expected authorization_token to be str or bytes")
+    if isinstance(authority_selector, bytes):
+      pass
+    elif isinstance(authority_selector, basestring):
+      authority_selector = authority_selector.encode()
+    else:
+      raise TypeError("expected authority_selector to be str or bytes")
+    credentials = ClientCredentials()
+    credentials.c_credentials = grpc.grpc_iam_credentials_create(
+        authorization_token, authority_selector)
+    credentials.references.append(authorization_token)
+    credentials.references.append(authority_selector)
+    return credentials
+
+  def __dealloc__(self):
+    if self.c_credentials != NULL:
+      grpc.grpc_credentials_release(self.c_credentials)
+
+
+cdef class ServerCredentials:
+  cdef grpc.grpc_server_credentials *c_credentials
+  cdef list references
+  def __cinit__(self):
+    self.c_credentials = NULL
+
+  @staticmethod
+  def ssl(pem_root_certs, pem_key_cert_pairs):
+    if pem_root_certs is None:
+      pass
+    elif isinstance(pem_root_certs, bytes):
+      pass
+    elif isinstance(pem_root_certs, basestring):
+      pem_root_certs = pem_root_certs.encode()
+    else:
+      raise TypeError("expected pem_root_certs to be str or bytes")
+    pairs = []
+    for pair in pem_key_cert_pairs:
+      pass
+    raise NotImplementedError("fuuuuuuuu")
+
+  #grpc_server_credentials *grpc_ssl_server_credentials_create(
+  #    const char *pem_root_certs,
+  #    grpc_ssl_pem_key_cert_pair *pem_key_cert_pairs,
+  #    size_t num_key_cert_pairs);
+
+  @staticmethod
+  def fake_transport_security():
+    credentials = ServerCredentials()
+    credentials.c_credentials = \
+        grpc.grpc_fake_transport_security_server_credentials_create()
+    return credentials
+
+  def __dealloc__(self):
+    if self.c_credentials != NULL:
+      grpc.grpc_server_credentials_release(self.c_credentials)
+
+
+cdef class CompletionQueue:
+  # MUST: all tags pushed onto this completion queue are Python objects or
+  # NULL.
+  # HIGHLY RECOMMENDED: all tags pushed onto this completion queue are
+  # OperationTag.
+  # TODO(atash) thread safety!
+  cdef grpc.grpc_completion_queue *c_completion_queue
+  # because shutdown isn't yet idempotent...
```

Comment:
More like, I don't know *how* Cython uses the GIL in each generated method (or, rather, I haven't found the documentation for it yet). If it by default holds the GIL for an entire method, then we're already fine. If it doesn't, then I need to add in mutexes. I'll find out about that more later - I was intended this pass mostly for style comments. :-P